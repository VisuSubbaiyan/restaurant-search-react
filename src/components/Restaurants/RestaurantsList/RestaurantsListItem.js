import React from 'react';
import PropTypes from 'prop-types';

const RestaurantsListItem = ({ name, location, selected, onClick }) => {
  const className = `restaurants-list-item ${selected ? 'selected' : ''}`;

  return (
    <li className={className} onClick={onClick}>
      <h3 className="name">{name}</h3>
      <address className="address">{`${location.address1}, ${location.city}`}</address>
    </li>
  );
};

RestaurantsListItem.propTypes = {
  name: PropTypes.string,
  location: PropTypes.object,
};

export default RestaurantsListItem;
