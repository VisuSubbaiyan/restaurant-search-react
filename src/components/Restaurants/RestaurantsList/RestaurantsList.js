import React from 'react';
import PropTypes from 'prop-types';

import RestaurantsListItem from './RestaurantsListItem';

const RestaurantsList = ({ total, restaurants, selectedId, onClick }) => (
  <div className="restaurants-list">
    <h3 className="total-result">{`${total} Restaurants`}</h3>
    <ul>
      {restaurants.map(
        restaurant => {
          const selected = restaurant.id === selectedId;
          // onClick is muted for selected item, to avoid unnecessary rerendering
          return (
            <RestaurantsListItem
              key={restaurant.id}
              {...restaurant}
              selected={selected}
              onClick={!selected ? () => onClick(restaurant.id) : () => null}
            />);
        })
      }
    </ul>
  </div>
);

RestaurantsList.propTypes = {
  restaurants: PropTypes.arrayOf(PropTypes.object)
}

export default React.memo(RestaurantsList);