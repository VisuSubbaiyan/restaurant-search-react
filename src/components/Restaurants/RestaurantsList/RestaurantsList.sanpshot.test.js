import React from 'react';

import RestaurantsList from './RestaurantsList';
import { createMountedComponent, mockFunction, sampleData } from '../../../utils/testUtils';

describe('<RestaurantsList /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <RestaurantsList {...sampleData} onClick={mockFunction} />
  });

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
  })
});