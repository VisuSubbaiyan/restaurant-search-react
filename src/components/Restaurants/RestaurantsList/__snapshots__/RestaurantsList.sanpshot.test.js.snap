// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantsList /> should render correctly according to props 1`] = `
"<div class=\\"restaurants-list\\">
  <h3 class=\\"total-result\\">34 Restaurants</h3>
  <ul>
    <li class=\\"restaurants-list-item selected\\">
      <h3 class=\\"name\\">Billy The Butcher</h3><address class=\\"address\\">Jungfernstieg 16 - 20, Hamburg</address>
    </li>
  </ul>
</div>"
`;
